### PR TITLE
update to operation hooks

### PR DESCRIPTION
**Pull request by hacksparrow**

### CODE REVIEW

**Comment by hacksparrow** [comment]
Guys, any idea why Travis is failing this?

**Comment by jakerella** [comment]
Looks like a [jshint warning](http://ci.strongloop.com/job/loopback-example-app.pr/8/label=ubuntu-64-0.10/artifact/checkstyle.xml), but the warning says line 36 needs a semicolon, which it already has.

I see no issue with merging this otherwise.

**Comment by rmg** [comment]
@slnode test please

**Comment by crandmck** [comment]
LGTM

File: common/models/location.js [hunk 1]
```diff
@@ -31,7 +31,10 @@ module.exports = function(RentalLocation) {
     geoService.geocode.apply(geoService, arguments);
   });
 
-  RentalLocation.beforeSave = function(next, loc) {
+  RentalLocation.observe('before save', function(ctx, next) {
+
+    var loc = ctx.instance;
```
Comment (bajtos) [inline common/models/location.js]: Note that `ctx.instance` may be undefined, e.g. when calling `updateAttributes`. For the purpose of this hook, I think it's ok to do `var loc = ctx.instance || ctx.data`

**Comment by bajtos** [comment]
One comment, the rest LGTM.

**Comment by bajtos** [comment]
I have fixed the problem myself and landed the patch via f5a8e5b. Thank you for the contribution!

**Pull request closed without merging**